### ISSUE TITLE
Add diffGrad optimizer

### ISSUE THREAD

**Issue by suvojit-0x55aa** [pytorch/pytorch#27052]
## 🚀 Feature

Add [diffGrad](https://arxiv.org/abs/1909.11015) optimizer to Pytorch.

**Comment by gtamba-cs** [comment]
I noticed that the author has released Pytorch code [here](https://github.com/shivram1987/diffGrad/blob/master/diffGrad.py) using the pytorch-cifar package for experiments which should make adaptability easier

**Comment by soumith** [comment]
We don't think this method should be in pytorch core, as opposed to your own personal repository or something like https://github.com/pytorch/contrib , at least not yet in time.

Our reservation is that we want to include methods that the community uses as a standard, or else the code maintenance problem balloons up for us.
We do show discretion based on what the paper shows as evidence, for example BatchNorm was included within weeks of it's publish date (in Torch).
In terms of rejected methods, we've rejected newly minted papers such as Swish ( #3260 , #3182 ), Yellowfin ( https://github.com/pytorch/pytorch/issues/1960 ) and many others, and rightly so, these haven't become standardized in the community (like LSTM / Transformer / BatchNorm).

If you have a differing opinion, let us know why, and we can re-think.

tl;dr: The paper doesn't show evidence that makes it a method that has obvious long-term success. If the paper does have long-term success in the field we will include it
